pkg/types/transactions_erc20.go

```diff
@@ -0,0 +1,1414 @@
+package types
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/threefoldfoundation/tfchain/pkg/config"
+	"github.com/threefoldtech/rivine/crypto"
+	"github.com/threefoldtech/rivine/pkg/encoding/rivbin"
+	"github.com/threefoldtech/rivine/pkg/encoding/siabin"
+	"github.com/threefoldtech/rivine/types"
+)
+
+const (
+	// TransactionVersionERC20Conversion defines the Transaction version
+	// for an ERC20ConvertTransaction, used to convert TFT into ERC20 funds.
+	TransactionVersionERC20Conversion types.TransactionVersion = iota + 208
+	// TransactionVersionERC20CoinCreation defines the Transaction version
+	// for an ERC20CoinCreationTransaction, used to convert ERC20 funds into TFT.
+	TransactionVersionERC20CoinCreation
+	// TransactionVersionERC20AddressRegistration defines the Transaction version
+	// for an TransactionVersionERC20AddressRegistration, used to register an ERC20 address,
+	// linked to an TFT address.
+	TransactionVersionERC20AddressRegistration
+)
+
+// These Specifiers are used internally when calculating a Transaction's ID.
+// See Rivine's Specifier for more details.
+var (
+	SpecifierERC20ConvertTransaction             = types.Specifier{'e', 'r', 'c', '2', '0', ' ', 'c', 'o', 'n', 'v', 'e', 'r', 't', ' ', 't', 'x'}
+	SpecifierERC20CoinCreationTransaction        = types.Specifier{'e', 'r', 'c', '2', '0', ' ', 'c', 'o', 'i', 'n', 'g', 'e', 'n', ' ', 't', 'x'}
+	SpecifierERC20AddressRegistrationTransaction = types.Specifier{'e', 'r', 'c', '2', '0', ' ', 'a', 'd', 'd', 'r', 'r', 'e', 'g', ' ', 't', 'x'}
+)
+
+var (
+	// ERC20ConversionMinimumValue defines the minimum value of TFT
+	// you can convert to ERC20 funds using the ERC20ConvertTransaction
+	ERC20ConversionMinimumValue = config.GetCurrencyUnits().OneCoin.Mul64(1000)
+)
+
+// ERC20AddressLength defines the length of the fixed-sized ERC20Address type explicitly.
+const ERC20AddressLength = 20
+
+// ERC20Address defines an ERC20 address as a fixed-sized byte array of length 20,
+// and is used in order to be able to convert TFT into tradeable tfchain ERC20 funds.
+type ERC20Address [ERC20AddressLength]byte
+
+// String returns this ERC20Address as a string.
+func (address ERC20Address) String() string {
+	return hex.EncodeToString(address[:])
+}
+
+// LoadString loads this ERC20Address from a hex-encoded string of length 40.
+func (address *ERC20Address) LoadString(str string) error {
```

Comment:
Right now this function only works if it is passed the actual hexchars, though most popular (if not all) erc20 tools (such as metamask) generally include the `0x` prefix in addresses. It feels like better UI if we optionally allow this as well, otherwise a lot of users will need to manually remove this prefix before entering an address